### PR TITLE
fix the bug of IE conditional comments

### PR DESCRIPTION
**Pull request by adon-at-work**
- also renamed the config name to avoid misunderstanding
- refactored config manipulations as code optimizations

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: src/context-parser.js [hunk 1]
```diff
@@ -408,9 +406,17 @@ function Parser (config, listeners) {
...
         }
-    } else {
-        config = self.config;
     }
+    config = self.config;    
+
+    // config defaulted to false
+    config.enableInputPreProcessing = (config.enableInputPreProcessing === true);
+    config.enableCanonicalization = (config.enableCanonicalization === true);
+    config.enableVoidingIEConditionalComments = (config.enableVoidingIEConditionalComments === true);
```
Comment (neraliu) [inline src/context-parser.js]: please create the corresponding PR to the html-purifier and secure-handlebars for renaming this conf param

File: src/context-parser.js [hunk 2]
```diff
@@ -968,7 +971,7 @@ function DisableIEConditionalComments(state, i){
         // for lazy conversion
         this._convertString2Array();
 
-        this.input.splice(i, 0, ' ');
+        this.input.splice(i + 1, 0, ' ');
```
Comment (neraliu) [inline src/context-parser.js]: what is the root cause of the fix for IE conditional comment issues?

Comment (adon-at-work) [inline src/context-parser.js]: Was a bug. The initial commit is to add a space before ]>. When i is incremented, ]> is encountered again, and that caused an endless loop.

File: src/context-parser.js (hunk 1 above)
Comment (adon-at-work) [inline src/context-parser.js]: Ok. Once this is merged.

File: src/context-parser.js (hunk 2 above)
Comment (neraliu) [inline src/context-parser.js]: got it, please add back the corresponding unit test for this case

Comment (adon-at-work) [inline src/context-parser.js]: See the last file change

File: src/context-parser.js (hunk 1 above)
Comment (adon-at-work) [inline src/context-parser.js]: PRs of both are ready

**Pull request merged**